toolchain/check/testdata/interface/require.carbon

```diff
@@ -247,16 +240,32 @@ interface Z(T:! type) {
   require impls 1;
 }
 
+// --- require_self_in_requirement.carbon
+library "[[@TEST_NAME]]";
+
+interface Y {
+  let Y1:! type;
+}
+
+//@dump-sem-ir-begin
+interface Z {
+  // Self can appear in a requirement.
+  require impls Y where .Y1 = Self;
+}
+//@dump-sem-ir-end
+
```

Comment:
We should probably also have a test that `Self` in a requirement isn't sufficient to satisfy the "`Self` must appear in the type or interface arguments" rule:
```
interface Z2 {
  require i32 impls Y where .Y1 = Self;
}
```
should give an error.